elks/include/linuxmt/config.h

```diff
@@ -90,6 +90,23 @@
 #define CONFIG_8018X_EB
 #endif
 
+#ifdef CONFIG_ARCH_SWAN
+#define MAX_SERIAL              1       /* max number of serial tty devices*/
+#define SETUP_VID_COLS          28      /* video # columns */
+#define SETUP_VID_LINES         18      /* video # lines */
+#define SETUP_CPU_TYPE          5       /* processor type 80186 */
+#define SETUP_MEM_KBYTES        128     /* base memory in 1K bytes */
+#define SETUP_XMS_KBYTES        0       /* xms memory in 1K bytes */
+#define SETUP_ROOT_DEV          0x0600  /* root device ROMFS */
+#define SETUP_ELKS_FLAGS        0       /* flags for root device type */
+#define SETUP_PART_OFFSETLO     0       /* partition offset low word */
+#define SETUP_PART_OFFSETHI     0       /* partition offset high word */
+#define SYS_CAPS                0       /* no XT/AT capabilities */
+#define UTS_MACHINE             "swan"
+#define SETUP_HEAPSIZE          32768
+#define CONFIG_MEM_SEGMENT      0x1000
```

Comment:
Since CONFIG_MEM_SEGMENT is brand new, please add a comment after it explaining its purpose, something like /* start segment for application memory heap */ or similar.